packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -459,6 +463,7 @@ export function SelectPanel({
           role: 'dialog',
           'aria-labelledby': titleId,
           'aria-describedby': subtitle ? subtitleId : undefined,
+          ...(variant === 'modal' ? {'data-variant': 'modal' /* override AnchoredOverlay */} : {}),
```

Comment:
this is a bit icky, we can't pass variant=anchored here because it overrides AnchoredOverlay :/